src/traces/image/defaults.js

```diff
@@ -34,14 +34,17 @@ module.exports = function supplyDefaults(traceIn, traceOut) {
     coerce('dx');
     coerce('dy');
 
+    var cm;
     if(traceOut._hasZ) {
         coerce('colormodel');
-        coerce('zmin', constants.colormodel[traceOut.colormodel].min);
-        coerce('zmax', constants.colormodel[traceOut.colormodel].max);
+        cm = constants.colormodel[traceOut.colormodel];
+        coerce('zmin', (cm.zminDflt || cm.min));
+        coerce('zmax', (cm.zmaxDflt || cm.max));
     } else if(traceOut._hasSource) {
-        traceOut.colormodel = 'rgba';
-        traceOut.zmin = constants.colormodel[traceOut.colormodel].min;
-        traceOut.zmax = constants.colormodel[traceOut.colormodel].max;
+        traceOut.colormodel = 'rgba256';
+        cm = constants.colormodel[traceOut.colormodel];
+        traceOut.zmin = (cm.zminDflt || cm.min);
+        traceOut.zmax = (cm.zmaxDflt || cm.max);
```

Comment:
At the moment, all the values that depend on the colormodels are kept in file `image/constants.js`. Hardcoding `zmin` and `zmax` here would mean duplicating information we already have in the codebase. I could remove `|| cm.min` and `|| cm.max` however since we'll never reach it in that block.

cc @archmoj

Reply:
> I could remove `|| cm.min` and `|| cm.max` however since we'll never reach it in that block.

Done in 66c58f6c8aa38505c1904210e1dc4f24bab3155c